server/conn_test.go

```diff
@@ -922,7 +922,9 @@ func TestHandleAuthPlugin(t *testing.T) {
 	drv := NewTiDBDriver(store)
 	srv, err := NewServer(cfg, drv)
 	require.NoError(t, err)
-	ctx := context.Background()
+	ctx := failpoint.WithHook(context.Background(), func(ctx context.Context, fpname string) bool {
+		return fpname == "github.com/pingcap/tidb/server/FakeAuthSwitch"
+	})
```

Comment:
Should move this change after line #956?

Reply:
And after line #976 disabled the failpoint, you should use `context.Background()` instead of this ctx?

Reply:
since the `ctx` with hook should only been used between failpoint enable and disable.